src/hub.js

```diff
@@ -538,24 +538,34 @@ document.addEventListener("DOMContentLoaded", async () => {
   window.APP.scene = scene;
   window.APP.hubChannel = hubChannel;
 
+  const preEmptVRMode = () => {
+    // If VR headset is activated, refreshing page will fire vrdisplayactivate
+    // which puts A-Frame in VR mode, so exit VR mode whenever it is attempted
+    // to be entered and we haven't entered the room yet.
+    console.log("Pre-emptively exiting VR mode.");
+
+    if (scene.is("vr-mode") && !scene.is("entered")) {
+      scene.exitVR();
+      return true;
+    }
+  };
+
   scene.addEventListener("enter-vr", () => {
+    if (preEmptVRMode()) return;
     document.body.classList.add("vr-mode");
 
     // Don't stretch canvas on cardboard, since that's drawing the actual VR view :)
     if (!isMobile || availableVREntryTypes.cardboard !== VR_DEVICE_AVAILABILITY.yes) {
       document.body.classList.add("vr-mode-stretch");
     }
-
-    if (!scene.is("entered")) {
-      // If VR headset is activated, refreshing page will fire vrdisplayactivate
-      // which puts A-Frame in VR mode, so exit VR mode whenever it is attempted
-      // to be entered and we haven't entered the room yet.
-      console.log("Pre-emptively exiting VR mode.");
-      scene.exitVR();
-    }
   });
 
-  scene.addEventListener("exit-vr", () => document.body.classList.remove("vr-mode"));
+  preEmptVRMode();
+
+  scene.addEventListener("exit-vr", () => {
+    document.body.classList.remove("vr-mode");
```

Comment:
Seems odd that we have to remove the `vr-mode` class. I thought this is what `exitVR` in aframe did: https://github.com/MozillaReality/aframe/blob/hubs/master/src/core/scene/a-scene.js#L376